sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala

```diff
@@ -660,6 +662,10 @@ case class ShowPartitionsCommand(
   override def run(sparkSession: SparkSession): Seq[Row] = {
     val catalog = sparkSession.sessionState.catalog
 
+    if (!catalog.tableExists(table)) {
+      throw new AnalysisException(s" Table does not exist")
```

Comment:
Can you include the table name in this message? Also please remove the random space in the beginning. Something like:
```
s"Table $table does not exist"
```